docs/design/sql/05-ddl.md

```diff
@@ -0,0 +1,305 @@
+  )
+  ```
+
+  Column names and types are inferred as:
+
+  ```sql
+  (
+    __key INT,
+    this VARCHAR
+  )
+  ```
+
+- Create an external table referencing an `IMap` with explicit schema.
+
+  ```sql
+  CREATE EXTERNAL TABLE persons (
+    id INT EXTERNAL NAME __key,
+    name VARCHAR
+  )
+  TYPE "com.hazelcast.IMap"
+  OPTIONS (
+    "serialization.key.format" 'java',
+    "serialization.key.java.class" 'java.lang.Integer',
+    "serialization.value.format" 'java',
+    "serialization.value.java.class" 'java.lang.String'
+  )
+  ```
+
+- Create an external table referencing an `IMap` with explicit schema and column external names.
+
+  ```sql
+  CREATE EXTERNAL TABLE persons (
+    key_id INT EXTERNAL NAME "__key.id",
+    value_id INT EXTERNAL NAME "this.id"
+  )
+  TYPE "com.hazelcast.IMap"
+  OPTIONS (
+    "serialization.key.format" 'java',
+    "serialization.key.java.class" 'com.hazelcast.Person',
+    "serialization.value.format" 'java',
+    "serialization.value.java.class" 'com.hazelcast.Person'
+  )
+  ```
+
+- Create an external table referencing an `IMap` with schema inferred from `Portable` class definitions.
+
+  ```sql
+  CREATE EXTERNAL TABLE persons
+  TYPE "com.hazelcast.IMap"
+  OPTIONS (
+    "serialization.key.format" 'portable',
+    "serialization.key.portable.factoryId" '1',
+    "serialization.key.portable.classId" '2',
+    "serialization.key.portable.classVersion" '3',
+    "serialization.value.format" 'portable',
+    "serialization.value.portable.factoryId" '4',
+    "serialization.value.portable.classId" '5',
+    "serialization.value.portable.classVersion" '6'
+  )
+  ```
+
+- Create an external table referencing an `IMap` with JSON as the serialization format.
+
+  ```sql
+  CREATE EXTERNAL TABLE persons (
+    id INT EXTERNAL NAME "__key.id",
+    name VARCHAR EXTERNAL NAME "this.name"
+  )
+  TYPE "com.hazelcast.IMap"
+  OPTIONS (
+    "serialization.key.format" 'json',
+    "serialization.value.format" 'json'
+  )
+  ```
+
+## 2 CREATE EXTERNAL TABLE AS
+
+`CREATE EXTERNAL TABLE AS` -- define a new external table from the results of a query
+
+### 2.1 Synopsis
+
+```
+CREATE [OR REPLACE] EXTERNAL TABLE [IF NOT EXISTS] table_name [ (
+ column_name [ EXTERNAL NAME column_external_name ] [, ... ]
+) ]
+TYPE table_type
+[ OPTIONS ( table_parameter table_parameter_value [, ... ] ) ]
+AS query
+```
+
+### 2.2 Description
+
+`CREATE EXTERNAL TABLE AS` registers a virtual table that references an external storage system. It fills the table with
+data computed by `SELECT` statement. The table columns have the names and data types associated with the output columns
+of the `SELECT` (except that column names can be overridden by an explicit list of names).
+
+The table is created in the `public` schema. The name of the table must be distinct from the name of any other table in
+the schema.
+
+### 2.3 Parameters
+
+**OR REPLACE**
+
+Replace table definition if one with the same name already exists.
```

Comment:
`DROP` does not clear nor delete the underlying storage system, why `OR REPLACE` should do that?

Reply:
Because it's going to load data into that table. I think `CREATE OR REPLACE` should be equivalent of `DROP` + `CREATE`, except that it's atomic. Also `CREATE TABLE .. AS SELECT` should be equivalent of `CREATE TABLE` + `INSERT ... SELECT`, but atomic. And `CREATE OR REPLACE .. AS SELECT` is `DROP` + `CREATE` + `INSERT .. SELECT`, but atomic. 

In our case the `CREATE + INSERT` part is not atomic, we should make this very clear. But I still think it might be confusing to the users to find out that the old data weren't removed. But thinking about it, even if you don't use the `OR REPLACE` part, the target object can still contain data.

So we should make it more clear that the `CREATE [OR REPLACE]` part only applies to the link and pre-existing data are not cleared. And also that it's not atomic.